### PR TITLE
feat: create tab component

### PR DESCRIPTION
**Pull request by ashinzekene**
#### What does this PR do
Creates tab component according to the [style guide](https://www.figma.com/file/wtojDaoY2PJED946ZMIvsX/OSCA-style-guide%2Fpattern)
 
#### How should this be manually tested
Run storybook and check the tab components

#### What are the relevant issues this PR belongs to
#23

#### Screenshots
image

**NOTE:** The bar/line at the top is not part of the component. It is from storybook

### CODE REVIEW

File: src/components/tab/tab.module.css [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+  text-transform: uppercase;
+  justify-content: center;
+  align-items: center;
+  text-align: center;
+  font-size: 14px;
+  /* Without this; Safari briefly fades the text color whilst ::active.*/
+  -webkit-fontsmoothing: subpixel-antialiased;
+}
+
+.tabContainer:hover {
```
Comment (BolajiAyodeji) [inline src/components/tab/tab.module.css]: Can you add some left and right padding to the hover states so the text stands out from the hover color?

File: src/css/colors.css [hunk 2]
```diff
@@ -0,0 +1,9 @@
+@value primaryColor: #283756;
```
Comment (BolajiAyodeji) [inline src/css/colors.css]: We already have colors setup in `tailwind`, see #2. 
Can you use colors from there or add to if you need more colors.

Or are you trying to make the component independent which makes sense?

File: src/components/tab/tabs.stories.js [hunk 3]
```diff
@@ -0,0 +1,110 @@
+import React, { useState } from "react";
+import { Tabs, Tab } from ".";
+
+export default { title: "Tabs" };
```
Comment (BolajiAyodeji) [inline src/components/tab/tabs.stories.js]: Please add the component name so the storybook docs addon can work effectively, see #41

```js
export default {
    title: 'Tabs',
    component: 'Tabs'
};
```

File: src/components/tab/tab.module.css (hunk 1 above)
Comment (ashinzekene) [reply src/components/tab/tab.module.css]: Oh yeah. Okay

File: src/css/colors.css (hunk 2 above)
Comment (ashinzekene) [reply src/css/colors.css]: Oh cool, I'd use that then. Didn't know. I'd update the other PRs too

File: src/components/tab/tab.module.css [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+}
+
+.tabsTab {
+  height: 100%;
+  position: relative;
+}
+
+.tabsTab:active,
+.tabsTab:focus {
+  outline: none;
```
[the remark this answers is not in the document]
Comment (ashinzekene) [reply src/components/tab/tab.module.css]: From the page

> 2. Style the element itself. You can remove the outline as long as you style the focused element differently (using a combination of color, background-color, border or text-decoration: underline for example). When using this approach, avoid using color alone as the only visual means of determining focus, as colorblind people may not be able to distinguish the focus state from the inactive state.

The focused tab already has this styling that differentiates it from other tabs
image

File: src/components/tab/tab.module.css [hunk 5]
```diff
@@ -0,0 +1,74 @@
+@value colors: "../../css/colors.css";
```
[the remark this answers is not in the document]
Comment (ashinzekene) [reply src/components/tab/tab.module.css]: The file no longer exists. I've used tailwind instead

[the remark this answers is not in the document]
Comment (BolajiAyodeji) [reply src/components/tab/tab.module.css]: I actually meant the `tab.module.css`

[the remark this answers is not in the document]
Comment (ashinzekene) [reply src/components/tab/tab.module.css]: Oh okay

[the remark this answers is not in the document]
Comment (ashinzekene) [reply src/components/tab/tab.module.css]: Done

**Comment by BolajiAyodeji** [comment]
@allcontributors please add @ashinzekene for code

**Pull request merged**